### PR TITLE
Addform

### LINKED ISSUE #37: Add offer screen
# User story

As a **customer**,
I want to **add details to the food**
so I can **share it.**

# Notes / Links

# Acceptance criteria

- [ ]
- [ ]
- [ ]
- [ ]
- [ ]

### PR DESCRIPTION
**Pull request by manuelpapa**
# User story

As a **customer**,
I want to **add details to the food**
so I can **share it.**

image

### CODE REVIEW

File: client/src/components/CategoryPicker.js [hunk 1]
```diff
@@ -1,12 +1,36 @@
...
-  color: #000000;
-  border: 0;
-  cursor: pointer;
+  position: relative;
+  select {
+    font-family: "SFUIreg";
+    color: var(--font-semi-dark);
+    font-size: 1em;
+    padding: 0.5em 3em 0.5em 1em;
+    margin: 0em;
```
Comment (manuelsanchez2) [inline client/src/components/CategoryPicker.js]: You can use just 0 without the em

File: client/src/components/DateTimePicker.js [hunk 2]
```diff
@@ -1,16 +1,29 @@
...
         selected={selectedDate}
-        onChange={(date) => setSelectedDate}
-      ></DatePicker>
+        onChange={(date) => setSelectedDate(date)}
+        dateFormat="dd.MM.yyyy"
+        isClearable="true"
+      />
+
+      <RangePicker
+        format="H:mm"
```
Comment (manuelsanchez2) [inline client/src/components/DateTimePicker.js:22]: This is cool! never seen that before!

File: client/src/components/Form.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
+import styled from "@emotion/styled";
+import React from "react";
+import { useForm } from "react-hook-form";
+
+export default function Form() {
+  const { register, handleSubmit, errors } = useForm();
+  const onSubmit = (data) => console.log(data);
+  console.log(errors);
```
Comment (manuelsanchez2) [inline client/src/components/Form.js:8]: If it already works, do not forget to remove the console.log

File: client/src/components/Form.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            min: 2,
+            maxLength: 15,
+            pattern: /^\S+@\S+$/i,
+          })}
+        />
+        <input
+          type="number"
+          placeholder="PLZ"
+          name="PLZ"
+          ref={register({ required: true, min: 5, maxLength: 6 })}
```
Comment (manuelsanchez2) [inline client/src/components/Form.js]: I guess we can use either min and max or minLenght and maxLenght, right?

File: client/src/components/PageLayout.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  input,
+  select {
+    border: solid 1px var(--font-semi-dark);
+    padding: 0.5em;
+  }
+  p,
+  input,
+  select {
+    font-size: 1em;
+    margin: 0em;
```
Comment (manuelsanchez2) [inline client/src/components/PageLayout.js]: Remove the em

File: client/src/pages/Categories.js [hunk 6]
```diff
@@ -67,18 +62,15 @@ export function Categories() {
...
-              <h2>Kategorie</h2>
-            </ListItem>
-          ))}
-        </List>
-      </Main>
-    </>
+    <PageLayout>
+      <List>
+        {results.map((result) => (
+          <ListItem key={result.id} href={`/offers/${result.id}`}>
```
Comment (manuelsanchez2) [inline client/src/pages/Categories.js:68]: oh shit git! this looks great!

**Review by manuelsanchez2 — APPROVED** [review]
lgtm! keep doing like this! I am looking forward to seeing the deployment!

File: client/src/components/CategoryPicker.js (hunk 1 above)
Comment (manuelpapa) [reply client/src/components/CategoryPicker.js]: bravo! so true.

File: client/src/components/Form.js (hunk 4 above)
Comment (manuelpapa) [reply client/src/components/Form.js]: Didn't knew that before, thank you!

File: client/src/pages/Categories.js (hunk 6 above)
Comment (manuelpapa) [reply client/src/pages/Categories.js:68]: 😄

File: client/src/components/Form.js (hunk 3 above)
Comment (manuelpapa) [reply client/src/components/Form.js:8]: Will do this in a later step!

**Pull request merged**